File: module_music/src/main/java/com/yyxnb/module_music/ui/MusicNetworkFragment.java
package com.yyxnb.module_music.ui;

import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.yyxnb.adapter.BaseViewHolder;
import com.yyxnb.adapter.MultiItemTypeAdapter;
import com.yyxnb.arch.annotations.BindRes;
import com.yyxnb.arch.annotations.BindViewModel;
import com.yyxnb.common_base.base.BaseFragment;
import com.yyxnb.lib_music.MusicPlayerManager;
import com.yyxnb.module_music.R;
import com.yyxnb.module_music.adapter.MusicNetWorkListAdapter;
import com.yyxnb.module_music.databinding.FragmentMusicNetworkBinding;
import com.yyxnb.module_music.db.MusicDatabase;
import com.yyxnb.module_music.viewmodel.MusicViewModel;

/**
 * 网络音乐.
 */
@BindRes(subPage = true)
public class MusicNetworkFragment extends BaseFragment {

    private FragmentMusicNetworkBinding binding;
    private RecyclerView mRecyclerView;
    private MusicNetWorkListAdapter mAdapter;

    @BindViewModel
    MusicViewModel mViewModel;

    @Override
    public int initLayoutResId() {
        return R.layout.fragment_music_network;
    }

    @Override
    public void initView(Bundle savedInstanceState) {
        binding = getBinding();
        mRecyclerView = binding.mRecyclerView;
    }

    @Override
    public void initViewData() {
        mAdapter = new MusicNetWorkListAdapter();
        mRecyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        mRecyclerView.setHasFixedSize(true);
        mRecyclerView.setAdapter(mAdapter);

        mAdapter.setOnItemClickListener(new MultiItemTypeAdapter.SimpleOnItemClickListener() {
            @Override
            public void onItemClick(View view, BaseViewHolder holder, int position) {
                super.onItemClick(view, holder, position);

                // 更新数据源
                MusicDatabase.getInstance().musicDao().deleteAll();
                MusicDatabase.getInstance().musicDao().insertItems(mAdapter.getData());
                MusicPlayerManager.getInstance().startPlayMusic(mAdapter.getData(), position);
            }
        });
    }

    @Override
    public void initObservable() {
        mViewModel.reqMusicData();

        mViewModel.getMusicData().observe(this, data -> {
            if (data != null && data.size() > 0) {
                mAdapter.setDataItems(data);
            }
        });
    }
}

File: module_music/src/main/java/com/yyxnb/module_music/ui/MusicNetworkFragment.kt
package com.yyxnb.module_music.ui

import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.yyxnb.adapter.BaseViewHolder
import com.yyxnb.adapter.MultiItemTypeAdapter
import com.yyxnb.arch.annotations.BindRes
import com.yyxnb.arch.annotations.BindViewModel
import com.yyxnb.common_base.base.BaseFragment
import com.yyxnb.lib_music.MusicPlayerManager
import com.yyxnb.module_music.R
import com.yyxnb.module_music.adapter.MusicNetWorkListAdapter
import com.yyxnb.module_music.bean.MusicBean
import com.yyxnb.module_music.databinding.FragmentMusicNetworkBinding
import com.yyxnb.module_music.db.MusicDatabase.Companion.instance
import com.yyxnb.module_music.viewmodel.MusicViewModel

/**
 * 网络音乐.
 */
@BindRes(subPage = true)
class MusicNetworkFragment : BaseFragment() {
    private var binding: FragmentMusicNetworkBinding? = null
    private var mRecyclerView: RecyclerView? = null
    private var mAdapter: MusicNetWorkListAdapter? = null

    @BindViewModel
    var mViewModel: MusicViewModel? = null
    override fun initLayoutResId(): Int {
        return R.layout.fragment_music_network
    }

    override fun initView(savedInstanceState: Bundle?) {
        binding = getBinding()
        mRecyclerView = binding!!.mRecyclerView
    }

    override fun initViewData() {
        mAdapter = MusicNetWorkListAdapter()
        mRecyclerView!!.layoutManager = LinearLayoutManager(context)
        mRecyclerView!!.setHasFixedSize(true)
        mRecyclerView!!.adapter = mAdapter
        mAdapter!!.setOnItemClickListener(object : MultiItemTypeAdapter.SimpleOnItemClickListener() {
            override fun onItemClick(view: View, holder: BaseViewHolder, position: Int) {
                super.onItemClick(view, holder, position)

                // 更新数据源
                instance!!.musicDao().deleteAll()
                instance!!.musicDao().insertItems(mAdapter!!.data)
                MusicPlayerManager.getInstance().startPlayMusic(mAdapter!!.data, position)
            }
        })
    }

    override fun initObservable() {
        mViewModel!!.reqMusicData()
        mViewModel!!.musicData.observe(this, { data: List<MusicBean?>? ->
            if (data != null && data.size > 0) {
                mAdapter!!.setDataItems(data)
            }
        })
    }
}